src/dmd/cppmanglewin.d

```diff
@@ -684,22 +684,138 @@ private:
             buf.writestring("?1");
             return;
         }
+        else if (sym.ident)
+        {
+            switch (sym.ident.toString)
+            {
+                case "opAssign":    buf.writestring("?4");  return;
+                case "opEquals":    buf.writestring("?8");  return;
+                case "opIndex":     buf.writestring("?A");  return;
+                case "opCall":      buf.writestring("?R");  return;
+                default:            break;
+            }
+        }
         if (TemplateInstance ti = sym.isTemplateInstance())
         {
+            auto id = ti.tempdecl.ident;
+            const(char)* ident = id.toChars;
+
+            // test for special symbols
+            int firstTemplateArg = 0;
+            int whichOp = -1;
+            switch (ti.name.toString)
+            {
+                case "opCast":
+                    buf.writestring("?B");
+                    return;
+                case "opUnary":     whichOp = 0;    break;
+                case "opBinary":    whichOp = 1;    break;
+                case "opOpAssign":  whichOp = 2;    break;
+                case "opAssign":    whichOp = 3;    break;
+                case "opEquals":    whichOp = 4;    break;
+                case "opIndex":     whichOp = 5;    break;
+                case "opCall":      whichOp = 6;    break;
+                default:            break;
+            }
+            if (whichOp != -1)
+            {
+                TemplateDeclaration td = ti.tempdecl.isTemplateDeclaration();
+                assert(td);
+                TemplateParameter tp = (*td.parameters)[0];
+                TemplateValueParameter tv = tp.isTemplateValueParameter();
+                if (tv)
+                {
+                    if (tv.valType.isString())
+                    {
+                        Expression exp = (*ti.tiargs)[0].isExpression();
+                        StringExp str = exp.toStringExp();
```

Comment:
Please use the operator, not the string representation of the operator.

Reply:
I'm not sure what this means?

Reply:
`exp.op`

Reply:
I don't understand... `exp.op == 121`, that is `TOK.string_`... That is, the expression is a StringExp, which I get on the next line.  
Is `toStringExp()` expensive?

Or are you saying there's some way for me to get the operator that the string represents without testing the instantiation string? How can I know what the string value of the first template arg is without evaluating it? What if someone instantiates with ctfe?